numba/parfor.py

```diff
@@ -2997,7 +3006,17 @@ def get_reduction_init(nodes):
         return 1, acc_expr.fn
     return None, None
 
-def get_reduce_nodes(name, nodes):
+def supported_reduction(x, func_ir):
+    if x.op == 'inplace_binop' or x.op == 'binop':
+        return True
+    if x.op == 'call':
+        callname = guard(find_callname, func_ir, x)
+        callname = tuple(i if i != '__builtin__' else 'builtins' for i in callname)
```

Comment:
What's the `else` branch in here for?

Reply:
@stuartarchibald We match against ('max', 'builtins') later on because that is what you get there for Python 3.  However, for Python 2, the normal thing you get here is ('max', '__builtin__').  This line just replaces __builtin_ with builtins.

Reply:
Ah thanks. I read this wrong and was assuming that it was a tuple like `(("function name 1", "builtins"), ("function name 2", "builtins"))`, which it's clearly not.